src/pybind/mgr/cephadm/module.py

```diff
@@ -2539,6 +2539,23 @@ def _apply(self, spec: GenericSpec) -> str:
     def apply_tuned_profiles(self, specs: List[TunedProfileSpec]) -> str:
         outs = []
         for spec in specs:
+
+            # ensure all the configured sysctl options are valid
+            if spec.settings is not None:
```

Comment:
I feel like this whole block here checking the settings ought to get its own function and we should probably have a unit test for it.

Reply:
moved to a new function and UT added for the same